src/cocotb/share/lib/vpi/VpiObj.cpp

```diff
@@ -63,48 +63,41 @@ int VpiArrayObjHdl::initialise(const std::string &name,
 
     /* After determining the range_idx, get the range and set the limits */
     vpiHandle iter = vpi_iterate(vpiRange, hdl);
-
-    s_vpi_value val;
-    val.format = vpiIntVal;
+    vpiHandle rangeHdl;
 
     if (iter != NULL) {
-        vpiHandle rangeHdl;
-        int idx = 0;
+        rangeHdl = vpi_scan(iter);
 
-        while ((rangeHdl = vpi_scan(iter)) != NULL) {
-            if (idx == range_idx) {
+#ifdef MODELSIM
+        for (int i = 0; i < range_idx; ++i) {
+            rangeHdl = vpi_scan(iter);
+            if (rangeHdl == NULL) {
                 break;
```

Comment:
I think this line is a different error. We *can* scan, but it's yielding less than the identified number of dimensions. So instead of "Unable to get range for indexable array", maybe it's "Unable to get range for all dimensions of indexable array"?